src/Tribe/Ajax/Dropdown.php

```diff
@@ -235,20 +235,33 @@ public function convert_children_to_array( $results ) {
 	 * Parses the Params coming from Select2 Search box
 	 *
 	 * @since  4.6
+	 * @since TBD Added an allow list of params to restrict the shape of the database queries.
 	 *
 	 * @param array<string|mixed> $params Params to overwrite the defaults
 	 *
 	 * @return object
 	 */
 	public function parse_params( $params ) {
-		$defaults = [
+		$defaults       = [
 			'search' => null,
 			'page'   => 0,
-			'args'   => [],
 			'source' => null,
 		];
+		$allowed_params = [
```

Comment:
Why not just do a list of allowed keys and `array_intersect_key`?

Reply:
I think it would be more complicated/lines of code. As you mentioned this does the defaults too, with `array_intersect_key` we would not get defaults right?

```php
        $allowed_params = [
            'search' => $params['search'] ?? null,
            'page'   => $params['page'] ?? 0,
            'source' => $params['source'] ?? null,
            'args'   => [ 'post_status' => 'publish' ],
        ];
```